### PR TITLE
exec: new package

### PR DESCRIPTION
**Pull request by vibhavp**
This change is [Reviewable](https://reviewable.io/reviews/go-interpreter/wagon/23)

### CODE REVIEW

File: exec/num.go [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+	vm.pushUint32(v1 / v2)
+}
+
+func (vm *VM) i32RemS() {
+	v2 := vm.popInt32()
+	v1 := vm.popInt32()
+	vm.pushInt32(v1 % v2)
+}
+
+func (vm *VM) i3RemU() {
```
Comment (sbinet) [inline exec/num.go]: 's/i3RemU/i32RemU/'

**Comment by vibhavp** [comment]
Also, there are a few testing errors related to some changes in the `disasm` package. WIll push a fix soon.

**Comment by vibhavp** [comment]
CI tests with Go 1.7 fail:
```
wagon_test.go:20: error running go vet:
		exec/internal/bits/bits.go:11: ^uint(0) might be too small for shift of 32
		exit status 1
```
However, this doesn't seem to be the case with 1.8.

**Comment by sbinet** [comment]
yeah... I was actually trying to devise some workaround for this...

**Comment by sbinet** [comment]
please cherry-pick 876f81a. it should work.

**Comment by sbinet** [comment]
that's a lot of files to review in one sitting...

could you first send a PR with only `exec/internal/compile`. it seems that package has manageable dependencies.

**Comment by vibhavp** [comment]
sure.

**Comment by sbinet** [comment]
thanks. BTW: traditionally, test data directories are called `testdata`, not `test_data`.
(this has a nice little optimization in the `go` toolchain: `go get ./...` won't inspect the `testdata` directories nor its children)

**Pull request closed without merging**